src/phoenix/server/api/types/TimeSeries.py

```diff
@@ -70,17 +71,24 @@ def _get_timeseries_data(
 ) -> List[TimeSeriesDataPoint]:
     metric_cls = metric.value
     data = dimension[dataset_role.value]
-    if issubclass(metric_cls, NullaryOperator):
-        metric_instance = metric_cls()
-    else:
-        assert issubclass(metric_cls, UnaryOperator)
-        metric_instance = metric_cls(operand=Column(dimension.name))
-    if issubclass(metric_cls, DriftOperator):
+    metric_instance = metric_cls()
+    if isinstance(metric_instance, UnaryOperator):
+        metric_instance = replace(
+            metric_instance,
+            operand=Column(dimension.name),
+        )
+    if isinstance(metric_instance, DriftOperator):
         ref_data = dimension[REFERENCE]
-        metric_instance.reference_data = pd.DataFrame({dimension.name: ref_data})
+        metric_instance = replace(
+            metric_instance,
+            reference_data=pd.DataFrame({dimension.name: ref_data}),
+        )
         if dimension.data_type is CONTINUOUS:
-            metric_instance.binning_method = binning.QuantileBinning(
-                reference_series=dimension[REFERENCE],
+            metric_instance = replace(
```

Comment:
:thought: no way to avoid the replace? Feels a tad dirty as it feels like we should be passing these in rather than mutating an instance.

Reply:
`replace` is actually not that bad, because it's actually creating a brand new instance (the old instance is frozen), and by doing so runs it through `__post_init__` at runtime, which lets us run validations if there are any